### PR TITLE
feat: support peek multi value for query args

### PR DESCRIPTION
**Pull request by a631807682**
#### What type of PR is this?

feat

#### Check the PR title.

- [x] This PR title match the format: \<type\>(optional scope): \<description\>
- [x] The description of this PR title is user-oriented and clear enough for others to understand.

#### (Optional) Translate the PR title into Chinese.
支持获取多个值的query参数

#### (Optional) More detail description for this PR(en: English/zh: Chinese).

en: 
`ctx.Query` cannot get multiple values of `a` from `a=1&a=2&b=1`.
`QueryArray` is provided in `gin` to get multiple values, but [rfc3986](https://www.rfc-editor.org/rfc/rfc3986#section-3.4) does not define the implementation of the server, and many frameworks are inconsistent when parsing repeated queries , some only use the first or last occurrence.
So I'm not sure it needs to be implemented in `RequestContext`, but I hope there is still a way to get it.
```go
 ctx.QueryArgs().PeekMulti("a")
```
zh(optional):
`ctx.Query` 无法从 `a=1&a=2&b=1` 获取 `a` 的多个值。
`gin`中提供了`QueryArray`获取多个值，但是 [rfc3986](https://www.rfc-editor.org/rfc/rfc3986#section-3.4) 没有定义服务端的实现，很多框架在解析的时候不一致，有些只使用第一个值或最后一个值。
所以我不确定它是否需要在`RequestContext`中实现，但我希望有办法获到它。
```go
 ctx.QueryArgs().PeekMulti("a")
```

### CODE REVIEW

**Comment by li-jin-gou** [comment]
refer to https://github.com/cloudwego/hertz/pull/569/files and do you have a real world example of this use?

**Comment by a631807682** [comment]
> refer to [#569 (files)](https://github.com/cloudwego/hertz/pull/569/files) and do you have a real world example of this use?

Do you mean I need to describe this scenario in a test like https://github.com/cloudwego/hertz/pull/569/files#diff-6b0ef71f167bc3e1767a55944dc08d8cfb025629780b5cfcceb1cbe21e668e21R577 ?
Or do I need to add relevant code samples in the PR's description?

**Comment by li-jin-gou** [comment]
> > refer to [#569 (files)](https://github.com/cloudwego/hertz/pull/569/files) and do you have a real world example of this use?
> 
> Do you mean I need to describe this scenario in a test like https://github.com/cloudwego/hertz/pull/569/files#diff-6b0ef71f167bc3e1767a55944dc08d8cfb025629780b5cfcceb1cbe21e668e21R577 ? Or do I need to add relevant code samples in the PR's description?

Reference code implementation

**Comment by li-jin-gou** [comment]
> > > > refer to [#569 (files)](https://github.com/cloudwego/hertz/pull/569/files) and do you have a real world example of this use?
> > > 
> > > 
> > > Do you mean I need to describe this scenario in a test like [#569 (files)](https://github.com/cloudwego/hertz/pull/569/files#diff-6b0ef71f167bc3e1767a55944dc08d8cfb025629780b5cfcceb1cbe21e668e21R577) ? Or do I need to add relevant code samples in the PR's description?
> > 
> > 
> > Reference code implementation
> 
> Sorry, I still don't get your point, can you tell me what aspect of PR implementation I need to refer to?
> 
> 1. Is it about api design?
>    We need to provide this api on RequestContext? There are two different considerations here, and the current implementation is the first one, please let me know if there is any problem.
>    [Added ctx.QueryMulti() gofiber/fiber#288 (comment)](https://github.com/gofiber/fiber/pull/288#issuecomment-618739905)
>    [Implement QueryArray and PostFormArray methods gin-gonic/gin#570](https://github.com/gin-gonic/gin/pull/570)
> 2. Are there performance concerns?
>    QueryArgs will only be parsed once, this method is not much different from [GetQuery](https://github.com/cloudwego/hertz/blob/develop/pkg/app/context.go#L1229)
> 3. Did not describe the real usage scenario?
>    Do I need to add tests for multiple parameter requests to describe it?
> 
> The above is all my guesswork, I'd like some clear advice to let me know what I should do. 🙏

ok

File: pkg/protocol/args.go [hunk 1]
```diff
@@ -379,6 +379,17 @@ func (a *Args) PeekExists(key string) (string, bool) {
 	return peekArgStrExists(a.args, key)
 }
 
+// PeekMulti returns all the arg values for the given key.
+func (a *Args) PeekMulti(key string) [][]byte {
```
Comment (welkeyever) [inline pkg/protocol/args.go:383]: May unify the naming style with headers API: `PeekAll()`

File: pkg/protocol/args.go [hunk 2]
```diff
@@ -379,6 +379,17 @@ func (a *Args) PeekExists(key string) (string, bool) {
...
+	var values [][]byte
+	a.VisitAll(func(k, v []byte) {
+		if string(k) == key {
```
Comment (welkeyever) [inline pkg/protocol/args.go:386]: If only used in line 386, `bytesconv.B2s(k)` may be a better choice - less allocation

Comment (a631807682) [reply pkg/protocol/args.go]: Thanks for your review, I checked B2s and found that it has updated its implementation in go1.20 
https://go.dev/src/strings/builder.go#L45
https://github.com/golang/go/issues/53003
https://github.com/valyala/fasthttp/blob/master/b2s_new.go

Comment (welkeyever) [reply pkg/protocol/args.go]: About the updated version in go1.20, it is welcome to submit another PR to do so~

**Comment by li-jin-gou** [comment]
users needs to use this PR and I will CR it today

**Review by li-jin-gou — APPROVED** [review]

**Review by Duslia — APPROVED** [review]

**Review by FGYFFFF — APPROVED** [review]

**Review by welkeyever — APPROVED** [review]

**Pull request merged**